src/Microsoft.OData.Cli/GenerateCommand.cs

```diff
@@ -252,10 +248,38 @@ private TServiceConfig GetServiceConfiguration<TServiceConfig>(GenerateOptions g
                 configUserSettings = configFileData?.ExtendedData;
             }
 
-            var namespacePrefix = string.IsNullOrEmpty(generateOptions.NamespacePrefix) ? configUserSettings?.NamespacePrefix : generateOptions.NamespacePrefix;
-            var excludedSchemaTypes = generateOptions.ExcludedSchemaTypes?.Split(",", StringSplitOptions.RemoveEmptyEntries).Select(type => type.Trim()).ToList();
-            var excludedBoundOperations = generateOptions.ExcludedBoundOperations?.Split(",", StringSplitOptions.RemoveEmptyEntries).Select(operation => operation.Trim()).ToList();
-            var excludedOperationImports = generateOptions.ExcludedOperationImports?.Split(",", StringSplitOptions.RemoveEmptyEntries).Select(import => import.Trim()).ToList();
+            // Empty namespace prefix commandline option should indicate not to apply any namespace prefix
+            var useNamespacePrefix = false;
+            string namespacePrefix = null;
+            if (!(generateOptions.NamespacePrefix != null && generateOptions.NamespacePrefix.All(char.IsWhiteSpace)))
```

Comment:
If this if statement a bit hard to parse because of the double negation (`!` and `!=`). I think the following would be easier to read:
```suggestion
            if (generateOptions.NamespacePrefix == null && !generateOptions.NamespacePrefix.All(char.IsWhiteSpace))
```
We check if it's null or all whitespace. Why do we need this check though?

Reply:
@habbes I rewrote the logic so it's much easier to follow. There's also the comment indicating what the logic is trying to accomplish